### PR TITLE
bpo-17909: Document that json.load can accept a binary IO

### PR DESCRIPTION
**Pull request by asottile**
https://bugs.python.org/issue17909

### CODE REVIEW

File: Doc/library/json.rst [hunk 1]
```diff
@@ -262,6 +262,11 @@ Basic Usage
    .. versionchanged:: 3.6
       All optional parameters are now :ref:`keyword-only <keyword-only_parameter>`.
 
+   .. versionchanged:: 3.6
+      *fp* can now be a :term:`file-like object` whose ``.read()`` method
```
Comment (serhiy-storchaka) [inline Doc/library/json.rst]: Why not use just ``` :term:`binary file` ```?

Comment (asottile) [reply Doc/library/json.rst]: I can go with that, was mostly matching the verbiage above (`load` after all only cares that the object has a `.read()` method)

File: Doc/library/json.rst [hunk 2]
```diff
@@ -262,6 +262,10 @@ Basic Usage
    .. versionchanged:: 3.6
       All optional parameters are now :ref:`keyword-only <keyword-only_parameter>`.
 
+   .. versionchanged:: 3.6
+      *fp* can now be a :term:`binary file`. The input encoding should be
+      UTF-8, UTF-16 or UTF-32.
```
Comment (vstinner) [inline Doc/library/json.rst:267]: IMHO the feature should also be documented in the body of the function documentation, not also in this versionchanged note.

Comment (asottile) [reply Doc/library/json.rst:267]: agreed, added!

**Comment by asottile** [comment]
@vstinner @serhiy-storchaka look good now?

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**